File: 프로그래머스/0/181939. 더 크게 합치기/더 크게 합치기.java

class Solution {
    private int concatenate_two_number(int num1, int num2) {
        int digits = 0;
        int tmp_num2 = num2;
        while (tmp_num2 != 0) {
            tmp_num2 /= 10;
            digits += 1;
        }
        return num1 * (int)Math.pow(10, digits) + num2;
    }

    public int solution(int a, int b) {
        int a_to_b = concatenate_two_number(a, b);
        int b_to_a = concatenate_two_number(b, a);
        int answer = (a_to_b > b_to_a) ? a_to_b : b_to_a;
        return answer;
    }
}

File: 프로그래머스/0/181939. 더 크게 합치기/더 크게 합치기.py
def solution(a, b):
	answer: int = 0
	tmp_a: int = a
	tmp_b: int = b
	a_digits: int = 0
	b_digits: int = 0
	while (tmp_a != 0):
		tmp_a //= 10
		a_digits += 1
	while (tmp_b != 0):
		tmp_b //= 10
		b_digits += 1
	a_to_b: int = a * int(10 ** b_digits) + b
	b_to_a: int = b * int(10 ** a_digits) + a
	answer = max(a_to_b, b_to_a)
	return answer